database/migrations/2025_01_16_180229_create_temp_projects_table.php

```diff
@@ -0,0 +1,85 @@
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+
+        // We are going to add a workflow to show the uploaded project data and validation result in front end.
+        // User can find project record easily, make corrections in excel file and then import again.
+        //
+        // This table is created for storing temporary projects data in the uploaded excel file.
+        //
+        // When all data are correct in excel file, user can import the excel file as real projects records.
+        // At this point, all related temp_projects records can be removed.
+        //
+        // Consider the purpose of this table, below design are applied:
+        // 1. All columns are either string or text (user can key in anything in any numeric field)
+        // 2. All columns are nullable (user can leave any field empty)
+        // 3. Records are related to a project import, which links to a user
+        // 4. Related records will be removed when user re-upload the excel file or user finialised to save the excel file
+
+        Schema::create('temp_projects', function (Blueprint $table) {
+            $table->id();
+
+            $table->foreignId('temp_project_import_id')
+                ->references('id')
+                ->on('temp_project_imports')
+                ->cascadeOnUpdate()
+                ->cascadeOnDelete();
+
+            $table->foreignId('organisation_id')
+                ->references('id')
+                ->on('organisations')
+                ->cascadeOnUpdate()
+                ->cascadeOnDelete();
+
+            $table->foreignId('portfolio_id')
+                ->references('id')
+                ->on('portfolios')
+                ->cascadeOnUpdate()
+                ->cascadeOnDelete();
+
+            $table->string('code')->nullable();
+            $table->string('name')->nullable();
+            $table->string('category')->nullable();
+            $table->text('description')->nullable();
+            $table->string('currency')->nullable();
+            $table->string('exchange_rate')->nullable();
+            $table->string('exchange_rate_eur')->nullable();
+            $table->string('budget')->nullable();
+            $table->string('uses_only_own_funds')->nullable();
+            $table->string('main_recipient')->nullable();
+            $table->string('start_date')->nullable();
+            $table->string('end_date')->nullable();
+            $table->string('geographic_reach')->nullable();
+            $table->string('continent_1')->nullable();
+            $table->string('continent_2')->nullable();
+            $table->string('region_1')->nullable();
+            $table->string('region_2')->nullable();
+            $table->string('country_1')->nullable();
+            $table->string('country_2')->nullable();
+            $table->string('country_3')->nullable();
+            $table->string('country_4')->nullable();
```

Comment:
This shouldn't be a fixed set of columns. Users might be adding region_3, country_5 etc, as per the instructions in the Excel template. We should be able to present any/all continents, regions and countries back to the user in the TempProjectCrudController's index page so that they are not concerned that their location data is missing from the import.

This probably means setting up many-many relationships for each location level to TempProject. (Or making them Json, but that's probably more complicated? And hopefully we can copy the data handling from the main ProjectImport process)